### DISCUSSION TITLE
Multibyte support for STR::snake/kebab methods

### DISCUSSION THREAD

**Discussion by zoryamba** [laravel/framework#51660] [Ideas]
`Str::snake` still use `ucwords` which doesn't support multibyte strings.
It's the last use of `ucwords` in `Str` class. It was removed previously from `Str::studly` in #40468
This seems to be an obvious bug for me, but PR to fix this (#51626) was closed with "No plans" reasoning. 

So I want to raise a discussion, should `Str::snake` work correct with multibyte strings? 

Actual behavior:
```php
Str::snake('ÖffentlicheÜberraschungen'); // öffentlicheüberraschungen
Str::snake('ŻółtaŁódka') // żółtałódka
```
Expected behavior:
```php
Str::snake('ÖffentlicheÜberraschungen'); // öffentliche_überraschungen
Str::snake('ŻółtaŁódka') // żółta_łódka
```

**Comment by shaedrich** [comment]
Yeah, multibyte support would be the standard. Having non-multibyte strings with multibyte support has no disadvantages to my knowledge, so it's hard to argue against this accept for laziness in terms of implementation maybe, but this is a very weak argument.
